components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -141,6 +141,30 @@ class SystemEngineSession : EngineSession() {
             }
         } ?: throw IllegalStateException("System engine not initialized")
 
+    /**
+     * See [EngineSession.setDesktopMode]
+     */
+    override fun setDesktopMode(enable: Boolean, reload: Boolean) {
+        currentView()?.settings?.userAgentString?.let {
+            val webSettings = currentView()!!.settings
```

Comment:
I think we should try to avoid using `!!`. Let's use a nested `let` instead? 
```
currentView()?.let { view ->
   view.settings?.userAgentString?.let {
   }
}
```

Reply:
That's a good point. Interestingly, I didnt' realize that `currentView` is the only nullable call so the second one disappears.